issue-16 include compound names in plot_volcano if names present

delfarahalireza opened the pull request:
### Description of your changes
Added an ```if``` statement to the ```geom_point``` call of the ```ggplot``` in order to include compound names on the claman::plot_volcano output if compoundName column is present in the input data. This would enable visualizing compound names on an interactive volcano plot with ```plotly::ggplotly(plot_volcano())```.

### Issue ticket number and link
#16 

### Type of change
- [ ] New feature (non-breaking change which adds functionality)

### (If applicable) How has this been tested?
tested with the following:
- input data that contains a comoundName
- input data that does not contain a compoundName

shackett commented on R/differential_expression.R:
```diff
@@ -294,31 +294,33 @@ plot_volcano <- function(
     ) %>%
     ggplot(aes_string(x = effect_var)) +
-    geom_point(aes(y = p.value.trans, color = is_discovery)) +
+    {if ("compoundName" %in% colnames(regression_significance))
```
This is fine to me since it will allow for plotly compatibility but won't affect the static plot. If you wanted to add labels in for the static plot that would probably be useful and I would recommend passing names or groupIds to label in as an optional argument

shackett commented on R/differential_expression.R:
```diff
@@ -294,31 +294,33 @@ plot_volcano <- function(
+    } +
     facet_wrap(~term, scales = "free_x") +
     scale_x_continuous("Effect size") +
-    scale_y_continuous(expression(-log[10] ~ "pvalue")) +
```
The labs call should be equivalent to the scale_y_continuous option but you removed the nice expression. Please restore "expression(-log[10] ~ "pvalue")" for the y-axis unless you have a good reason.

shackett reviewed (changes requested):
looks good aside from y-axis label change.

delfarahalireza commented:
Thanks for reviewing @shackett 

re: label for the static plot, do you suggest adding compound labels for all of the compounds? I thought that would overcrowd the plot.

re: ```scale_y_continuous``` with plotly the y-axis label gets printed as -log[10] ~ "pvalue" (but label is -log10 pvalue on the static version) that's why I replaced it with ```ylab```. Please see the image below:

<img width="878" alt="Screen Shot 2023-07-17 at 2 49 03 PM" src="https://github.com/calico/claman/assets/55253794/166015c1-93e6-48eb-a0b7-a16556f14a5d">

I will restore the ```scale_y_continuous(expression(-log[10] ~ "pvalue"))``` call for the function and I can overwrite the label when calling the plotting function, something like:

```plotly::ggplotly(plot_volcano_local(...)  + scale_y_continuous("-log10 pvalue"))``` which overwrites the ```scale_y_continuous(expression(-log[10] ~ "pvalue"))``` call within the function:

<img width="878" alt="Screen Shot 2023-07-17 at 2 58 05 PM" src="https://github.com/calico/claman/assets/55253794/54beeb30-7fb2-4800-8c2f-e44ad5c57d1a">

shackett reviewed (approved):
lgtm

For static plots I was thinking something that would allow for annotating specific features like this:
![age_x_lifespan_volcano](https://github.com/calico/claman/assets/508404/e749857b-63e6-4726-83cf-555070729281)

Its fine to tackle it in a separate issue (or just don't worry about it; up to you)

shackett commented:
@delfarahalireza you should also add your info to https://github.com/calico/claman/blob/main/DESCRIPTION since you've done enough for "aut" author role

delfarahalireza commented:
@shackett added a new input ```feature_labels``` to the function and added ```geom_text(aes(label = ifelse(compoundName %in% feature_labels, compoundName, ""), y = p.value.trans))``` to the ggplot call. Also roxygenized and added my info to the DESCRIPTION file as you suggested. I'll merge the PR if you approve and roxygenation looks good to you.

shackett commented:
@delfarahalireza lgtm; merge away!
